ui/app/src/components/PasswordTextField/PasswordTextField.tsx

```diff
@@ -38,13 +38,24 @@ const PasswordTextField = React.forwardRef<HTMLDivElement, PasswordTextFieldProp
   const { visibilitySwitch = true, initialVisible = false } = props;
   const { formatMessage } = useIntl();
   const [showPassword, setShowPassword] = useState(initialVisible);
-  const handleClickShowPassword = () => setShowPassword(!showPassword);
+  const inputRef = useRef<HTMLInputElement>();
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+    inputRef.current.focus();
+    setTimeout(() => {
+      const inputLength = inputRef.current.value.length;
+      inputRef.current.setSelectionRange(inputLength, inputLength);
+    }, 0);
```

Comment:
Why?

Reply:
When re-setting the focus the cursor was set back to position 0, so I'm setting it back to the latest position. Though, it needs a bit of time, hence the timeout